learn.py

```diff
@@ -0,0 +1,130 @@
+# Based on:
+# https://github.com/yunjey/pytorch-tutorial/blob/master/tutorials/02-intermediate/bidirectional_recurrent_neural_network/main.py
+
+import torch
+import torch.nn as nn
+from torch.utils import data
+import torchvision
+import torchvision.transforms as transforms
+import numpy as np
+import torch.nn.functional as F
+
+# Device configuration
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+
+# Bidirectional recurrent neural network (many-to-one)
+class BiRNN(nn.Module):
+    def __init__(self, input_size, hidden_size, num_layers):
+        super(BiRNN, self).__init__()
+        self.hidden_size = hidden_size
+        self.num_layers = num_layers
+        self.lstm = nn.LSTM(input_size, hidden_size, num_layers,
+                            batch_first=True, bidirectional=True, dropout=0.2)
+        self.lin = nn.Linear(256, 1000)
+        self.softmax = nn.Softmax(dim=1)
+        self.sigmoid = nn.Sigmoid()
+
+    def forward(self, x):
+        # Set initial states
+        h0 = torch.zeros(self.num_layers*2, x.size(0), self.hidden_size)\
+            .to(device)  # 2 for bidirection
+        c0 = torch.zeros(self.num_layers*2, x.size(0), self.hidden_size)\
+            .to(device)
+
+        # Forward propagate LSTM
+        # out: tensor of shape (batch_size, seq_length, hidden_size*2)
+        out, _ = self.lstm(x, (h0, c0))
+
+        # Decode the hidden state of the last time step
+        out = out[:, -1, :]
+        out = self.lin(out)
```

Comment:
This isnt in the original model description, but I see no other way to get input of size 1000 when the hidden rnn layer is of size 256.

Reply:
Resolved internally